frontend/src/components/DietPlan/DietPlan.tsx

```diff
@@ -1,24 +1,43 @@
-import { View, ImageBackground, ScrollView, TouchableOpacity, Text } from "react-native";
-import { useRef, useState } from "react";
-import dietPlan from "../../constants/dietPlan.json";
+import { View, ImageBackground, ScrollView,  Text, StyleSheet } from "react-native";
+import { useEffect, useRef, useState } from "react";
 import logoBlack from "../../../assets/images/avihu-logo-black.png";
-import CarbTable from "./CarbTable";
-import ProteinTable from "./ProteinTable";
 import useHideTabBarOnScroll from "@/hooks/useHideTabBarOnScroll";
+import { useDietPlanApi } from "@/hooks/useDietPlanApi";
+import { useUserStore } from "@/store/userStore";
+import { IDietPlan } from "@/interfaces/DietPlan";
+import MealContainer from "./MealContainer";
+import { Colors } from "@/constants/Colors";
+import NativeIcon from "../Icon/NativeIcon";
+import FABGroup from "../ui/FABGroup";
+import MenuItemModal from "./MenuItemModal";
 
-export default function DietPlan() {
-  const UI_TYPES = {
-    STANDARD: "STANDARD",
-    CARBS: "CARBS",
-    PROTEIN: "PROTEIN",
-  };
 
+export default function DietPlan() {
   const { handleScroll } = useHideTabBarOnScroll();
+  const currentUser = useUserStore((state) => state.currentUser);
+  const {getDietPlanByUserId}=useDietPlanApi()
 
-  const [meals, setMeals] = useState(dietPlan.meals);
-  const [uiView, setUiView] = useState(UI_TYPES.STANDARD);
+  const [dietPlan,setDietPlan]=useState<IDietPlan|null>(null)
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [isFabOpen,setIsFabOpen]=useState(false)
+  const [selectedFoodGroup,setSelectedFoodGroup]=useState<string |null>(null)
   const scrollRef = useRef(null);
 
+  const openModal=(foodGroup:string)=>{
```

Comment:
name of function does not accurately represent what is happening. 

Try to find a better name